### PR TITLE
Feat : Trace_transaction RPC call implementation

### PR DESCRIPTION
**Pull request by Tbelleng**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature ✅

## What is the new behavior?

Add trace_transaction RPC call and modify trace_block_transactions ✅
In this new implementation, we are going to separate transactions from a block into 2 vectors,
the first one contains transactions that we dont want to trace but we still need to re-execute them in order to
get the correct state update. 

The second vector contains transactions that we want to re-execute and trace.

⚠️ So for trace_block_transactions method, we are gonna re-execute and trace all transactions within a block, but concerning trace_transactions method, we will only target and put the transactions needed into the second vector to get its trace.

## Does this introduce a breaking change?

No ❌

### CODE REVIEW

File: crates/client/rpc/src/lib.rs
Comment (jbcaron) [inline crates/client/rpc/src/lib.rs:1]: is it really necessary to use an intermediate variable?

File: crates/pallets/starknet/src/tests/re_execute_transactions.rs
Comment (jbcaron) [inline crates/pallets/starknet/src/tests/re_execute_transactions.rs:1]: is it really necessary to use an intermediate variable?

File: crates/pallets/starknet/src/simulations.rs [hunk 1]
```diff
@@ -229,4 +248,45 @@ impl<T: Config> Pallet<T> {
...
+                    Self::execute_user_transaction(tx.clone(), chain_id, block_context, execution_config)
+                }
+                UserOrL1HandlerTransaction::L1Handler(tx, _fee) => {
+                    Self::execute_message(tx.clone(), chain_id, block_context, execution_config)
+                }
+            })
+            .collect();
+
+        let mut execution_infos = Vec::with_capacity(exec_transactions.len());
+        for result in exec_transactions {
```
Comment (jbcaron) [inline crates/pallets/starknet/src/simulations.rs]: is it possible to use a `.map` directly?

**Review by jbcaron — APPROVED** [review]

File: crates/client/rpc/src/lib.rs
Comment (Tbelleng) [reply crates/client/rpc/src/lib.rs:1]: Here the function changed in order to take 2 args as params, the first vector its tx that we dont want to trace and the second one is the ones we want to trace, so i added an empty vector for your function in order to fit the new function definiton

File: crates/pallets/starknet/src/simulations.rs (hunk 1 above)
Comment (Tbelleng) [reply crates/pallets/starknet/src/simulations.rs]: Good suggestion, im gonna try to 🤝

File: crates/pallets/starknet/src/tests/re_execute_transactions.rs
Comment (Tbelleng) [reply crates/pallets/starknet/src/tests/re_execute_transactions.rs:1]: Same comment as above, its now mandatory to use 2 vectors of transactions in order to get a proper re-execution

**Review by antiyro — APPROVED** [review]

**Pull request merged**